internal/resources/tanzukubernetescluster/schema.go

```diff
@@ -320,6 +326,32 @@ var NodePoolSpecSchema = &schema.Schema{
 			},
 			ReplicasKey: ReplicasSchema,
 			OSImageKey:  OSImageSchema,
+			AutoscalingKey: {
+				Type:        schema.TypeList,
+				Description: "Autoscaling config on the node pool",
+				MaxItems:    1,
+				Computed:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						AutoscalingEnabledKey: {
+							Type:        schema.TypeBool,
+							Description: "Enable autoscaling.",
+							Required:    true,
+						},
+						AutoscalingMinCountKey: {
+							Type:        schema.TypeInt,
+							Description: "The minimum number of nodes for autoscaling.",
+							Required:    true,
```

Comment:
is it still required if the AutoscalingEnabledKey is false?

Reply:
It is required when autoscaling is enabled, so it has to be marked as 'required', customer may input any value when autoscaling is disabled.